zio-http/src/main/scala/zhttp/service/server/ServerChannelInitializer.scala

```diff
@@ -1,19 +1,33 @@
 package zhttp.service.server
 
+import io.netty.handler.codec.http.HttpObjectDecoder.{
+  DEFAULT_MAX_CHUNK_SIZE,
+  DEFAULT_MAX_HEADER_SIZE,
+  DEFAULT_MAX_INITIAL_LINE_LENGTH,
+}
 import io.netty.handler.codec.http.HttpServerKeepAliveHandler
 import zhttp.core._
-import zhttp.service.{HTTP_KEEPALIVE_HANDLER, HTTP_REQUEST_HANDLER, OBJECT_AGGREGATOR, SERVER_CODEC_HANDLER}
+import zhttp.service._
 
 /**
  * Initializes the netty channel with default handlers
  */
 @JSharable
-final case class ServerChannelInitializer(httpH: JChannelHandler, maxSize: Int) extends JChannelInitializer[JChannel] {
+final case class ServerChannelInitializer(httpH: JChannelHandler, hhtpMfdH: JChannelHandler, maxSize: Int)
+    extends JChannelInitializer[JChannel] {
   override def initChannel(channel: JChannel): Unit = {
     channel
       .pipeline()
-      .addLast(SERVER_CODEC_HANDLER, new JHttpServerCodec)
+      // Without increasing the DEFAULT_MAX_CHUNK_SIZE uploads are rather slow. Probably because Netty is chunking up messages
+      // itself. With a file upload, that causes lots of overhead. A factor >8 (resulting in 65K chunks) is ignored so it seems.
+      .addLast(
+        SERVER_CODEC_HANDLER,
+        new JHttpServerCodec(DEFAULT_MAX_INITIAL_LINE_LENGTH, DEFAULT_MAX_HEADER_SIZE, DEFAULT_MAX_CHUNK_SIZE * 8),
+      )
       .addLast(HTTP_KEEPALIVE_HANDLER, new HttpServerKeepAliveHandler)
+      // This handler should always be in front of the `JHttpObjectAggregator` or we won't be able to
+      // decode multipart/form-data requests correctly.
+      .addLast(MULTIPART_FORMDATA_HANDLER, hhtpMfdH)
```

Comment:
We should add this handler conditionally ie. 
1. If the request is of multipart type (this is easy to check thru headers)
2. The user actually wants to parse it as a multipart request. It's possible that the user wants to send a 403 error or something else even before the data is uploaded (There is a way to check this also).

Reply:
Thanks for reviewing!

I see your points. Regarding 1), we let Netty detect whether it's a multipart because they have a built-in decoder already which probably also covers some edge cases. The multipart handler is then removed as soon as we know it's not a multipart request [here](https://github.com/dream11/zio-http/blob/99f61ab80488c458a25462166fc2507323ca5d93/zio-http/src/main/scala/zhttp/service/server/ServerMultipartRequestHandler.scala#L72) and [here](https://github.com/dream11/zio-http/blob/99f61ab80488c458a25462166fc2507323ca5d93/zio-http/src/main/scala/zhttp/service/server/ServerMultipartRequestHandler.scala#L84).

Removing a handler is much easier than adding at exactly the right position in the chain. Especially the latter is important as long as we have the aggregator handler.

Curious to see the details on 2)!